src/Features/Core/Portable/Shared/IDocumentSupportsFeatureService.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Shared
+{
+    internal interface IDocumentSupportsFeatureService : IWorkspaceService
+    {
+        bool SupportsCodeFixes(Document document);
+        bool SupportsRefactorings(Document document);
+        bool SupportsRename(Document document);
+        bool SupportsNavigationToAnyPosition(Document document);
+    }
+
+
+    [ExportWorkspaceService(typeof(IDocumentSupportsFeatureService), ServiceLayer.Default), Shared]
+    internal class DefaultDocumentSupportsFeatureService : IDocumentSupportsFeatureService
+    {
+        public bool SupportsCodeFixes(Document document)
```

Comment:
do we actually need one that accept Document for these? I think @dibarbet actually removed one that accept document and explicitly changed to text buffer for these APIs?

Reply:
I don't need those for my purpose, they were included just because I wanted to expose the same APIs in Feature layer. @dibarbet Any particular reason that those shouldn't be available for Document?

Reply:
I don't see any reason for them not to be available - it was changed from document -> text buffer because we had the text buffer first in almost all the cases the interface was being used, so we were often just getting the document for no reason.